tests/integration/long/test_ssl_topology_change.py

```
@@ -0,0 +1,567 @@
+                        timeout=5,
+                    )
+                    recovered = True
+                    log.info(f"Driver recovered after {attempt + 1} attempts")
+                    break
+                except Exception:  # noqa: BLE001
+                    time.sleep(2)
+
+            # The test PASSES if we DON'T get bad_fd_errors.
+            # If we DO get them, that confirms the bug reproduction.
+            assert not bad_fd_errors, (
+                f"python-driver issue #614 reproduced: Got 'Bad file descriptor' errors "
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)